projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-admin-menu.php

```diff
@@ -294,36 +294,50 @@ private function prepare_menu_item_icon( $icon ) {
 	 * @return string
 	 */
 	private function prepare_menu_item_url( $url, $parent_slug = '' ) {
-		// Calypso URLs need the base removed so they're not interpreted as external links.
-		if ( 0 === strpos( $url, 'https://wordpress.com' ) ) {
-			$url = str_replace( 'https://wordpress.com', '', $url );
-		} else {
-			$menu_hook   = get_plugin_page_hook( $url, $parent_slug );
-			$menu_file   = wp_parse_url( $url, PHP_URL_PATH ); // Removes query args to get a file name.
-			$parent_file = wp_parse_url( $parent_slug, PHP_URL_PATH );
+		// External URLS.
+		if ( preg_match( '/^https?:\/\//', $url ) ) {
+			// Allow URLs pointing to WordPress.com.
+			if ( 0 === strpos( $url, 'https://wordpress.com' ) ) {
```

Comment:
Should we try to be more strict to catch use-cases such as `https://wordpress.commerce.malicious-site.com`, here and below for Jetpack?

Reply:
https://github.com/Automattic/jetpack/pull/18759/commits/815eff6d5168966a86425a81300c9c97edd8e1de